### DISCUSSION TITLE
Duplicate .EXE files?

### DISCUSSION THREAD

**Discussion by gaiking-uk** [ImageMagick/ImageMagick#6247] [Help] [answered]
### ImageMagick version

ImageMagick-7.1.0-10-portable-Q16-HDRI-x64.zip

### Operating system, version and so on

Windows 11

### Description

I have generally opted to use ```Magick.exe``` in all my commands/scripts, however on reading a few answers on sites like [stackoverflow.com](https://stackoverflow.com/questions/9853325/how-to-convert-a-svg-to-a-png-with-imagemagick), I noticed sometimes people would use another binary, such as ```convert.exe``` for example.

Looking into the ImageMagick folder, I can see there are 9 different .exe files, however interestingly they all have the **EXACT** same filesize (to the byte!) and created/modified time 🤔

![ImageMagick - Identical EXEs (2)](https://user-images.githubusercontent.com/37931190/231156569-d81256c6-9659-4f38-b45d-a19d7811bd04.jpg)
### ❓ **Can I keep ```magick.exe``` and delete the other 8 *'copies'*?**

***

![ImageMagick - Identical EXEs (1)](https://user-images.githubusercontent.com/37931190/231156564-74cff229-5437-458f-b266-25ffc89981d9.jpg)

**Comment by snibgo** [comment]
On Windows, the `fc` utility will tell you if they are identical.

You can delete any `.exe` programs that you don't use, of course. IM v7 has been out for a few years, with the `magick` program. I do not recommend using `convert` for anything expect old scripts that rely on the old v6 syntax and can't be updated for some reason. (Most of my pages at http://im.snibgo.com have been updated to use `magick`.)

For v6 `convert`, use `magick`.

For the others such as `compare`, use `magick compare` etc.

**Comment by gaiking-uk** [comment]
Thanks for the quick reply!

I haven't used FC before so not *entirely* sure on if I have used it correctly but it generated a long series of hex codes -- comparing the two hex's side-by-side (assuming that's what you're meant to do), this would suggest the files **ARE** different...
```
Comparing files magick.exe and convert.exe
00000130: FF FA
00000180: 8A 96
00000181: 27 B8
00000182: 7D 7C
0161084C: FF FA
017C0C0B: 03 02
017C0C1A: F4 F3
017C0C1E: F0 EF
017C0C71: 98 86
...
```

NB: I did also generate SHA1 hashes of the files and these were different too (however I am not a hashing expert and wasn't sure if the hash comparison could be thrown off by something minor like a single line of code being different to generate different help/man info, etc(🤔?)

***

I appreciate the info/advice about the ```convert.exe```, with that in mind I might rephase my question slightly if that helps...

### ❓Can I use ```magick.exe``` for **all** IM commands/functions?... or are there some specific commands/conversions that only work using one of the other .exe files specifically (e.g. ```mogrify.exe```)?

<br/>
<br/>
<br/>

Am assuming that older scripts shared on [stackoverflow.com](https://stackoverflow.com/questions/48177348/gif-animation-with-imagemagick-and-many-images) can be converted to use ```magick.exe``` with something like the following(?)...
|  | Code |
| ----- | ----- |
| **OLD** | ❌ ```         mogrify.exe -path yourpath2/newdirectory -format gif -resize 20% *.jpg``` |
| **NEW** | ✅ ```magick.exe  -mogrify -path yourpath2/newdirectory -format gif -resize 20% *.jpg``` |
| **OLD** | ❌ ```         convert.exe -delay 20 *.gif -loop 0 animate.gif``` |
| **NEW** | ✅ ```magick.exe  -convert -delay 20 *.gif -loop 0 animate.gif``` |

**Comment by fmw42** [comment, marked as answer]
On IM 7, use magick in place of convert. But for all the other tools preface them with magick. So magick identify, magick mogrify, etc.  Just not magick convert. Only magick.

Generally, magick will do the same as the old convert in IM 6. But IM 7 has some new features that are not back ported to IM 6. See the porting guide at https://imagemagick.org/script/porting.php#cli

**Comment by gaiking-uk** [comment]
Thanks @fmw42 (and again to @snibgo) think that covers all the bases!

PS - I've only been using IM since v7 so hopefully won't have much conversion to do (except the odd script online that I want to adapt/re-use) and I feel pretty happy deleted the 8x other 'copies' of ```magick.exe``` as sounds like they are only included for legacy purposes / to support scripts that were written years ago and referenced those files instead of magick.exe, and aren't needed nowadays 👍🏼
